SCAutolib/models/gui.py

```diff
@@ -0,0 +1,304 @@
+from SCAutolib import run, logger
+
+import os
+from time import sleep, time
+import cv2
+import numpy as np
+import pandas as pd
+from io import StringIO
+import pytesseract
+import keyboard
+import uinput
+import functools
+
+
+class Screen():
+    """Captures the screenshots."""
+
+    def __init__(self, directory: str):
+        self.directory = directory
+        self.screenshot_num = 1
+
+    def screenshot(self, timeout: float = 30):
+        """Runs ffmpeg to take a screenshot.
+
+        :return: filename of the screenshot
+        :rtype: str
+        """
+
+        logger.debug(f"Taking screenshot number {self.screenshot_num}")
+
+        filename = f'{self.directory}/{self.screenshot_num}.png'
+        t_end = time() + timeout
+        captured = False
+
+        while time() < t_end and not captured:
+            out = run(['ffmpeg', '-hide_banner', '-y', '-f',
+                       'kmsgrab', '-i', '-', '-vf', 'hwdownload,format=bgr0',
+                       '-frames', '1', '-update', '1',
+                       filename], check=False, print_=False)
+
+            if out.returncode == 0:
```

Comment:
Is it enough to check returncode? Could it happen that returncode is 0 and screenshot is not saved?

Reply:
As far as I know, if the return code is 0, the screenshot has been captured successfully.